nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/AbstractKafkaProcessorLifecycelTest.java

```diff
@@ -0,0 +1,253 @@
+            assertTrue(e.getMessage().contains("'topic' is invalid because topic is required"));
+        }
+
+        runner.setProperty(ConsumeKafka.TOPIC, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+
+        runner.setProperty(ConsumeKafka.TOPIC, "  ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(ConsumeKafka.TOPIC, "blah");
+
+        runner.removeProperty(ConsumeKafka.CLIENT_ID);
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("invalid because client.id is required"));
+        }
+
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "   ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "ghj");
+
+        runner.setProperty(PublishKafka.KERBEROS_PRINCIPLE, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(PublishKafka.KERBEROS_PRINCIPLE, "  ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+    }
+
+    /*
+     * The goal of this test is to validate the correctness of
+     * AbstractKafkaProcessor's implementation of onTrigger() in a highly
+     * concurrent environment. That is:
+     * - upon processing failures (e.g., unhandled exceptions), the target Kafka
+     *   resource is reset (closed and re-created)
+     * - no FlowFile is unaccounted for. FlowFiles left in the queue and FlowFiles
+     *   in Success relationship = testCount
+     * - failed executions that did not result in the call to close/reset summed with
+     *   verified calls to close should equal total request failed
+     */
+    @Test
+    public void validateLifecycleCorrectnessWithProcessingFailures() throws Exception {
```

Comment:
@olegz @joewitt I do see the same test failure on my side and also the one below:

````
Running org.apache.nifi.processors.kafka.pubsub.ConsumeKafkaTest
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 52.39 sec <<< FAILURE! - in org.apache.nifi.processors.kafka.pubsub.ConsumeKafkaTest
validateGetAllMessagesWithProvidedDemarcator(org.apache.nifi.processors.kafka.pubsub.ConsumeKafkaTest)  Time elapsed: 1.112 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[????????????]-3> but was:<[????????????]-3>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.apache.nifi.processors.kafka.pubsub.ConsumeKafkaTest.validateGetAllMessagesWithProvidedDemarcator(ConsumeKafkaTest.java:191)
````
But maybe something is needed on my side to support Russian and Japanese characters?